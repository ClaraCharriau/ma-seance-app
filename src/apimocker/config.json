{
  "mockDirectory": "src/apimocker/",
  "quiet": false,
  "port": "7878",
  "latency": 50,
  "logRequestHeaders": false,
  "webServices": {
    "auth": {
      "mockFile": "users.json",
      "latency": 20,
      "verbs": ["post"]
    },
    "verify": {
      "mockFile": "verify.json",
      "latency": 20,
      "verbs": ["post"]
    }
  }
}
