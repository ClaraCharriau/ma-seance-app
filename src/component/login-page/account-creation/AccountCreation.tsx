import { FormEvent, useState } from 'react';
import { useAuth } from '../../../hooks/useAuth';
import style from '../Login.module.css';

interface AccountCreationProps {
    onSignUpClick: () => void;
}

const AccountCreation = (props: AccountCreationProps) => {
    const { onSignUpClick } = props;
    const { createUserAccount, checkUserExists } = useAuth();

    const [pseudo, setPseudo] = useState('');
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [passwordBis, setPasswordBis] = useState('');
    const [pseudoError, setPseudoError] = useState('');
    const [emailError, setEmailError] = useState('');
    const [passwordError, setPasswordError] = useState('');
    const [verifyError, setVerifyError] = useState('');

    const submitHandler = (e: FormEvent<HTMLFormElement>): void => {
        e.preventDefault();
        createAccount();
    };

    const createAccount = async (): Promise<void> => {
        const userExists = await checkUserExists(email);
        if (!userExists) {
            if (isFormValid()) {
                await createUserAccount(pseudo, email, password);
                onSignUpClick();
            } else {
                setVerifyError("Une erreur s'est produite.");
            }
        } else {
            setVerifyError('Un compte existe déjà avec cette adresse email.');
        }
    };

    const isFormValid = (): boolean => {
        setPseudoError('');
        setEmailError('');
        setPasswordError('');

        if ('' === pseudo) {
            setPseudoError('Choisissez un pseudo.');
            return false;
        }

        if (pseudo.length < 4) {
            setPseudoError('Votre pseudo doit contenir plus de 4 caractères.');
            return false;
        }

        if ('' === email) {
            setEmailError('Entrez votre adresse email.');
            return false;
        }

        if (!/^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/.test(email)) {
            setEmailError("Merci d'entrer une adresse email valide.");
            return false;
        }

        if ('' === password) {
            setPasswordError('Entrez votre mot de passe');
            return false;
        }

        if (passwordBis !== password) {
            setPasswordError('Les mots de passe doivent être identiques.');
            return false;
        }

        if (password.length < 7) {
            setPasswordError('Le mot de passe doit contenir au moins 8 caractères');
            return false;
        }

        return true;
    };

    return (
        <form className={style.loginForm} onSubmit={event => submitHandler(event)}>
            <h1 className={style.creationTitle}>Créer un compte</h1>

            {/* Pseudo */}
            <div className={style.inputContainer}>
                <label htmlFor="pseudo" className={style.inputLabel}>
                    Pseudo *
                </label>
                <input
                    type="text"
                    id="pseudo"
                    name="pseudo"
                    value={pseudo}
                    onChange={event => setPseudo(event.target.value)}
                    className={pseudoError ? style.errorInput : ''}
                    required
                />
                <label className={style.error}>{pseudoError}</label>
            </div>

            {/* E-mail adress */}
            <div className={style.inputContainer}>
                <label htmlFor="email" className={style.inputLabel}>
                    Adresse e-mail *
                </label>
                <input
                    type="mail"
                    id="email"
                    name="email"
                    value={email}
                    onChange={event => setEmail(event.target.value)}
                    className={emailError && style.errorInput}
                    required
                />
                <label className={style.error}>{emailError}</label>
            </div>

            {/* Password */}
            <div className={style.inputContainer}>
                <label htmlFor="password" className={style.inputLabel}>
                    Mot de passe *
                </label>
                <input
                    type="password"
                    id="password"
                    name="password"
                    value={password}
                    onChange={event => setPassword(event.target.value)}
                    className={passwordError && style.errorInput}
                    required
                />
                <label className={style.error}>{passwordError}</label>
            </div>

            {/* Password verification */}
            <div className={style.inputContainer}>
                <label htmlFor="passwordBis" className={style.inputLabel}>
                    Confirmez le mot de passe *
                </label>
                <input
                    type="password"
                    id="passwordBis"
                    name="passwordBis"
                    value={passwordBis}
                    onChange={ev => setPasswordBis(ev.target.value)}
                    className={passwordError && style.errorInput}
                    required
                />
                <label className={style.error}>{verifyError}</label>
            </div>

            <button type="submit" className={style.orangeBtn} onClick={createAccount}>
                S'inscrire
            </button>
        </form>
    );
};

export default AccountCreation;
