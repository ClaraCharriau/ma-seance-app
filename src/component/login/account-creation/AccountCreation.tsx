import { useState } from "react";
import { useAuth } from "../../../hooks/useAuth";

interface AccountCreationProps {
  onSignUpClick: () => void;
}

const AccountCreation = (props: AccountCreationProps) => {
  const { onSignUpClick } = props;
  const { createUserAccount, checkUserExists } = useAuth();

  const [pseudo, setPseudo] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [passwordBis, setPasswordBis] = useState("");
  const [pseudoError, setPseudoError] = useState("");
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");
  const [verifyError, setVerifyError] = useState("");

  const createAccount = async () => {
    let userExists = await checkUserExists(email);
    if (!userExists) {
      if (isFormValid()) {
        await createUserAccount(pseudo, email, password);
        onSignUpClick();
      } else {
        setVerifyError("Une erreur s'est produite.");
      }
    } else {
      setVerifyError("Un compte existe déjà à cette adresse email.");
    }
  };

  const isFormValid = (): boolean => {
    setPseudoError("");
    setEmailError("");
    setPasswordError("");

    if ("" === pseudo) {
      setPseudoError("Choisissez un pseudo.");
      return false;
    }

    if ("" === email) {
      setEmailError("Entrez votre adresse email.");
      return false;
    }

    if (!/^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/.test(email)) {
      setEmailError("Merci d'entrer une adresse email valide.");
      return false;
    }

    if ("" === password) {
      setPasswordError("Entrez votre mot de passe");
      return false;
    }

    if (password.length < 7) {
      setPasswordError("Le mot de passe doit contenir au moins 8 caractères");
      return false;
    }

    return true;
  };

  return (
    <section>
      <h2>Créer un compte</h2>

      {/* Pseudo */}
      <div className={"inputContainer"}>
        <label>Pseudo</label>
        <input
          value={pseudo}
          placeholder="Votre pseudo"
          onChange={(ev) => setPseudo(ev.target.value)}
          className={"inputBox"}
        />
        <label className="errorLabel">{pseudoError}</label>
      </div>

      {/* E-mail adress */}
      <div className={"inputContainer"}>
        <label>Adresse e-mail</label>
        <input
          value={email}
          placeholder="Votre email"
          onChange={(ev) => setEmail(ev.target.value)}
          className={"inputBox"}
        />
        <label className="errorLabel">{emailError}</label>
      </div>

      {/* Password */}
      <div className={"inputContainer"}>
        <label>Mot de passe</label>
        <input
          value={password}
          placeholder="Votre mot de passe"
          onChange={(ev) => setPassword(ev.target.value)}
          className={"inputBox"}
        />
        <label className="errorLabel">{passwordError}</label>
      </div>

      {/* Password verification */}
      <div className={"inputContainer"}>
        <label>Mot de passe</label>
        <input
          value={passwordBis}
          placeholder="Votre mot de passe"
          onChange={(ev) => setPasswordBis(ev.target.value)}
          className={"inputBox"}
        />
      </div>

      <div className={"inputContainer"}>
        <input
          className={"inputButton"}
          type="button"
          onClick={createAccount}
          value={"create account"}
        />
      </div>
      <label className="errorLabel">{verifyError}</label>
      <button onClick={createAccount} type="submit">S'inscrire</button>
    </section>
  );
};

export default AccountCreation;
