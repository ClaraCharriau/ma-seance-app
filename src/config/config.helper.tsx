/* eslint-disable @typescript-eslint/no-implicit-any-catch */
/* 
    Set configuration file depending on the environment.
    dev.json is the default configuration file.
*/
let config = require('./dev.json');

if (process.env.REACT_APP_ENV === 'staging') {
    try {
        const stagingConfig = require('./staging.json');
        config = { ...config, ...stagingConfig };
    } catch (error) {
        console.error('Could not load staging configuration file', error);
    }
}

export default config;
