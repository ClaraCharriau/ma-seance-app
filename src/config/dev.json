{
    "services" : {
        "prefix": "http://localhost:7878",
        "auth": {
            "verify": "/verify",
            "login": "/login",
            "registrations": "/registrations"
        }
    },
    "tmdbImgPath" : {
        "medium": "https://image.tmdb.org/t/p/w500",
        "large": "https://image.tmdb.org/t/p/w1280"
    }
}