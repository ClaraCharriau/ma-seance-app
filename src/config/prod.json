{
    "services" : {
        "prefix-url": "http://localhost:7878"
    }
}