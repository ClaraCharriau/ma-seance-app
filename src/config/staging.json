{
    "services": {
        "auth": {
            "verify": "http://localhost:3030/maseance/api/v1/auth/verify",
            "login": "http://localhost:3030/maseance/api/v1/auth/login",
            "registrations": "http://localhost:3030/maseance/api/v1/auth/registrations"
        },
        "users": {
            "users": "http://localhost:3030/maseance/api/v1/users/",
            "favTheaters": "/favorite-theaters",
            "favMovies": "/watchlist-movies",
            "screenings": "/screenings"
        },
        "theaters": {
            "theaters": "http://localhost:8080/maseance/api/v1/theaters",
            "screenings": "/screenings"
        },
        "screenings": {
            "screenings": "/screenings"
        },
        "movies": {
            "movies": "http://localhost:8080/maseance/api/v1/movies",
            "currently": "/currently",
            "screenings": "/screenings"
        }
    }
}
