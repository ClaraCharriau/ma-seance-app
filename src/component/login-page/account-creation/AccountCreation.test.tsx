/* eslint-disable  @typescript-eslint/no-empty-function */
import { act, fireEvent, render, waitFor } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import { axiosInstance } from '../../../client/axios.config';
import mockUser from '../../../mocks/users/users.json';
import AccountCreation from './AccountCreation';

describe('AccountCreation component tests', () => {
    let axiosMock: MockAdapter;

    beforeEach(() => {
        axiosMock = new MockAdapter(axiosInstance);
    });
    afterEach(() => {
        axiosMock.reset();
    });
    it('should render account creation component', () => {
        // Given
        const mockSignUpClickCallback = () => {
            console.log('click');
        };

        // When
        const component = render(<AccountCreation onSignUpClick={mockSignUpClickCallback} />);

        // Then
        expect(component.getByText('Créer un compte')).toBeInTheDocument();
        expect(component.getByLabelText('Pseudo *')).toBeInTheDocument();
        expect(component.getByLabelText('Adresse e-mail *')).toBeInTheDocument();
        expect(component.getByLabelText('Mot de passe *')).toBeInTheDocument();
        expect(component.getByLabelText('Confirmez le mot de passe *')).toBeInTheDocument();
        expect(component.getByText("S'inscrire")).toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });

    it('should create account', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        axiosMock.onPost('/registrations').reply(200, mockUser);
        const { getByLabelText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');
        const passwordInput = getByLabelText('Mot de passe *');
        const confirmPasswordInput = getByLabelText('Confirmez le mot de passe *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: 'toto@mail.it' } });
            fireEvent.change(passwordInput, { target: { value: 'awesomePassword123' } });
            fireEvent.change(confirmPasswordInput, { target: { value: 'awesomePassword123' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(axiosMock.history.post.length).toBe(2);
            expect(axiosMock.history.post[0].url).toBe('/token');
            expect(axiosMock.history.post[0].data).toEqual('{"email":"toto@mail.it"}');
            expect(axiosMock.history.post[1].url).toBe('/registrations');
            expect(axiosMock.history.post[1].data).toEqual(
                '{"pseudo":"TonySoprano","email":"toto@mail.it","password":"awesomePassword123"}'
            );
        });
    });

    it('should set already existing account error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: true
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: 'test@mail.com' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Un compte existe déjà avec cette adresse email.')).toBeInTheDocument();
        });
    });

    it('should return false and set missing pseudo error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: '' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Choisissez un pseudo.')).toBeInTheDocument();
            expect(getByText("Une erreur s'est produite.")).toBeInTheDocument();
        });
    });

    it('should return false and set missing pseudo error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'luc' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Votre pseudo doit contenir plus de 4 caractères.')).toBeInTheDocument();
        });
    });

    it('should return false and set missing email error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: '' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Entrez votre adresse email.')).toBeInTheDocument();
        });
    });

    it('should return false and set invalid email error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: 'invalidmail' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText("Merci d'entrer une adresse email valide.")).toBeInTheDocument();
        });
    });

    it('should return false and set missing password error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');
        const passwordInput = getByLabelText('Mot de passe *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: 'toto@mail.com' } });
            fireEvent.change(passwordInput, { target: { value: '' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Entrez votre mot de passe')).toBeInTheDocument();
        });
    });

    it('should return false and set unmaching password error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');
        const passwordInput = getByLabelText('Mot de passe *');
        const confirmPasswordInput = getByLabelText('Confirmez le mot de passe *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: 'toto@mail.com' } });
            fireEvent.change(passwordInput, { target: { value: 'pass123' } });
            fireEvent.change(confirmPasswordInput, { target: { value: 'pass456' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Les mots de passe doivent être identiques.')).toBeInTheDocument();
        });
    });

    it('should return false and set longer password error', async () => {
        // Given
        axiosMock.onPost('/token').reply(200, {
            exists: false
        });
        const { getByLabelText, getByText, getByRole } = render(<AccountCreation onSignUpClick={() => {}} />);
        const pseudoInput = getByLabelText('Pseudo *');
        const emailInput = getByLabelText('Adresse e-mail *');
        const passwordInput = getByLabelText('Mot de passe *');
        const confirmPasswordInput = getByLabelText('Confirmez le mot de passe *');

        // When
        act(() => {
            fireEvent.change(pseudoInput, { target: { value: 'TonySoprano' } });
            fireEvent.change(emailInput, { target: { value: 'toto@mail.com' } });
            fireEvent.change(passwordInput, { target: { value: '123' } });
            fireEvent.change(confirmPasswordInput, { target: { value: '123' } });
        });
        act(() => {
            fireEvent.submit(getByRole('button', { name: "S'inscrire" }));
        });

        // Then
        await waitFor(() => {
            expect(getByText('Le mot de passe doit contenir au moins 8 caractères')).toBeInTheDocument();
        });
    });
});
export { };

