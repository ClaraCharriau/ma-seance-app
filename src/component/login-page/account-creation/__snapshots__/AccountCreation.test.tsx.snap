// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCreation component tests should render account creation component 1`] = `
<body>
  <div>
    <form
      class="loginForm"
    >
      <h1
        class="creationTitle"
      >
        Cr√©er un compte
      </h1>
      <div
        class="inputContainer"
      >
        <label
          class="inputLabel"
          for="pseudo"
        >
          Pseudo *
        </label>
        <input
          class=""
          id="pseudo"
          name="pseudo"
          required=""
          type="text"
          value=""
        />
        <label
          class="error"
        />
      </div>
      <div
        class="inputContainer"
      >
        <label
          class="inputLabel"
          for="email"
        >
          Adresse e-mail *
        </label>
        <input
          class=""
          id="email"
          name="email"
          required=""
          type="mail"
          value=""
        />
        <label
          class="error"
        />
      </div>
      <div
        class="inputContainer"
      >
        <label
          class="inputLabel"
          for="password"
        >
          Mot de passe *
        </label>
        <input
          class=""
          id="password"
          name="password"
          required=""
          type="password"
          value=""
        />
        <label
          class="error"
        />
      </div>
      <div
        class="inputContainer"
      >
        <label
          class="inputLabel"
          for="passwordBis"
        >
          Confirmez le mot de passe *
        </label>
        <input
          class=""
          id="passwordBis"
          name="passwordBis"
          required=""
          type="password"
          value=""
        />
        <label
          class="error"
        />
      </div>
      <button
        class="orangeBtn"
        type="submit"
      >
        S'inscrire
      </button>
    </form>
  </div>
</body>
`;
